Scripts/upload_test_results.sh

```
@@ -114,56 +114,73 @@ then
+echo "Result bundle found: ${result_bundles}" 2>&1
 
-# 5. Find the .xcresult bundle.
+# 5. Generate the JUnit test report.
+junit_report_directory=$(dirname "${result_bundles}")
 
-# We use ~+ to give us an absolute path (https://askubuntu.com/a/1033450)
-result_bundles=$(find ~+/fastlane/test_output/sdk -name '*.xcresult')
-if [[ -z $result_bundles ]]
+# Delete the JUnit report that `scan` generated; we need to re-generate it by running `trainer` again, this time with the :force_legacy_xcresulttool option (which according to https://github.com/fastlane/fastlane/issues/29614) you can't pass directly to `scan`.
+echo "Deleting existing JUnit reports." 2>&1
+rm -f ${test_report_directory}/**/*.junit
```

Shellcheck (0.10.0)

[warning] test_report_directory is referenced but not assigned.

(SC2154)